src/supertux/gameconfig.cpp

```diff
@@ -49,14 +49,22 @@ Config::Config() :
   tux_spawn_pos(),
   edit_level(),
   locale(),
-  keyboard_config(),
-  joystick_config(),
+  keyboard_configs(),
+  joystick_configs(),
   addons(),
   developer_mode(false),
   christmas_mode(false),
   transitions_enabled(true),
   repository_url()
 {
+  assert(numberofplayers > 0);
+
+  for(int i = 0; numberofplayers > i; ++i){
+    keyboard_configs.push_back(std::shared_ptr<KeyboardConfig>(
```

Comment:
Sine we use C++11 can you use std::make_shared? (https://en.cppreference.com/w/cpp/memory/shared_ptr/make_shared)